terraform/lang/security/ec2-imdsv1-optional.yaml

```diff
@@ -0,0 +1,18 @@
+rules:
+  - id: ec2-imdsv1-optional
+    languages:
+      - generic
+    message: |
+      AWS EC2 Instance allowing use of the IMDSv1
+    metadata:
+      references:
+        - https://aws.amazon.com/blogs/security/defense-in-depth-open-firewalls-reverse-proxies-ssrf-vulnerabilities-ec2-instance-metadata-service
+        - https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/instance#metadata-options
+    paths:
+      include:
+        - '*.tf'
+    patterns:
+      - pattern-inside: metadata_options ...
```

Comment:
I think the following may make more sense:

```
- pattern-inside: |
    metadata_options {
        ...
        ...
        ...
    }
```

Reply:
Improved the constraint.